packages/react/src/components/SplitButton/SplitButton.tsx

```diff
@@ -0,0 +1,204 @@
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onClick?: ComponentEventHandler<ButtonProps>
+
+  /**
+   * Called after user's click on a menu item.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props.
+   */
+  onMenuItemClick?: ComponentEventHandler<MenuItemProps>
+
+  /** Defines whether menu is displayed. */
+  open?: boolean
+
+  /** A split button can be formatted to show different levels of emphasis. */
+  primary?: boolean
+
+  /** A split button can be formatted to show different levels of emphasis. */
+  secondary?: boolean
+
+  /** Shorthand for the toggle button. */
+  toggleButton?: ShorthandValue<ButtonProps>
+}
+
+export interface SplitButtonState {
+  isFromKeyboard: boolean
+  open: boolean
+}
+
+class SplitButton extends AutoControlledComponent<WithAsProp<SplitButtonProps>, SplitButtonState> {
+  static create: Function
+
+  static displayName = 'SplitButton'
+
+  static className = 'ui-splitbutton'
+
+  static slotClassNames: SplitButtonSlotClassNames = {
+    toggleButton: `${SplitButton.className}__toggleButton`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    button: customPropTypes.itemShorthand,
+    defaultOpen: PropTypes.bool,
+    menu: PropTypes.oneOfType([
+      customPropTypes.itemShorthand,
+      PropTypes.arrayOf(customPropTypes.itemShorthandWithoutJSX),
+    ]),
+    onClick: PropTypes.func,
+    onMenuItemClick: PropTypes.func,
+    open: PropTypes.bool,
+    primary: customPropTypes.every([customPropTypes.disallow(['secondary']), PropTypes.bool]),
+    secondary: customPropTypes.every([customPropTypes.disallow(['primary']), PropTypes.bool]),
+    toggleButton: customPropTypes.itemShorthand,
+  }
+
+  static defaultProps = {
+    accessibility: splitButtonBehavior,
+    as: 'span',
+  }
+
+  static autoControlledProps = ['open']
+
+  getInitialAutoControlledState(): SplitButtonState {
+    return {
+      isFromKeyboard: false,
+      open: false,
+    }
+  }
+
+  renderComponent({
+    ElementType,
+    classes,
+    accessibility,
+    styles,
+    unhandledProps,
+  }): React.ReactNode {
+    const { button, menu, primary, secondary, toggleButton } = this.props
+
+    return (
+      <ElementType className={classes.root} {...accessibility.attributes.root} {...unhandledProps}>
+        {MenuButton.create(
+          {},
+          {
+            defaultProps: {
+              accessibility: accessibility.childBehaviors.menuButton,
+              menu,
+              styles: styles.button,
+              on: [],
+              open: this.state.open,
+              onOpenChange: (e: React.SyntheticEvent, { open }: PopupProps) => {
+                e.stopPropagation()
+                this.setState({ open })
+              },
+              trigger: Button.create(button, {
```

Comment:
JSX it quite complicated there. I suggest to move out `trigger`:

```jsx
Button.create(button, {
  defaultProps: {
    styles: styles.button,
    primary,
    secondary,
  },
})
```

```suggestion
              trigger
```